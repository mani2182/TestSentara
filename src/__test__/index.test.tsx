import { render, fireEvent, waitFor } from "@testing-library/react";
import SentaraImageCropper from "../index";

jest.mock("react-avatar-edit", () => () => <div data-testid="avatarMock" />);

describe("SentaraImageCropper", () => {
  it("should render the modal when `imageCrop` prop is true", () => {
    const { getByTestId } = render(
      <SentaraImageCropper
        imageCrop={true}
        isDialogVisible={() => {}}
        saveCropImage={() => {}}
        callCroppedImage={() => {}}
        shadingColor="black"
        backgroundColor="white"
        buttonName="Save"
        modalTitle="Crop Image"
      />
    );

    const modalContainer = getByTestId("modalContainer");
    expect(modalContainer).toBeInTheDocument();
  });

  it("should hide the modal when `imageCrop` prop is false", () => {
    const { queryByTestId } = render(
      <SentaraImageCropper
        imageCrop={false}
        isDialogVisible={() => {}}
        saveCropImage={() => {}}
        callCroppedImage={() => {}}
        shadingColor="black"
        backgroundColor="white"
        buttonName="Save"
        modalTitle="Crop Image"
      />
    );

    const modalContainer = queryByTestId("modalContainer");
    expect(modalContainer).toBeNull();
  });

  it("should call `isDialogVisible` when the modal is closed", () => {
    const mockIsDialogVisible = jest.fn();
    const { getByLabelText } = render(
      <SentaraImageCropper
        imageCrop={true}
        isDialogVisible={mockIsDialogVisible}
        saveCropImage={() => {}}
        callCroppedImage={() => {}}
        shadingColor="black"
        backgroundColor="white"
        buttonName="Save"
        modalTitle="Crop Image"
      />
    );

    const closeButton = getByLabelText("Close");
    fireEvent.click(closeButton);

    expect(mockIsDialogVisible).toHaveBeenCalledWith(false);
  });

  it("should call `saveCropImage` when the 'Save' button is clicked", () => {
    const mockSaveCropImage = jest.fn();
    const { getByTestId } = render(
      <SentaraImageCropper
        imageCrop={true}
        isDialogVisible={() => {}}
        saveCropImage={mockSaveCropImage}
        callCroppedImage={() => {}}
        shadingColor="black"
        backgroundColor="white"
        buttonName="Save"
        modalTitle="Crop Image"
      />
    );

    const modalFooterButton = getByTestId("modalFooterButton");
    fireEvent.click(modalFooterButton);

    expect(mockSaveCropImage).toHaveBeenCalledWith(false);
  });

  it("should call `callCroppedImage` when the image is cropped", () => {
    const mockCallCroppedImage = jest.fn();
    const { getByTestId } = render(
      <SentaraImageCropper
        imageCrop={true}
        isDialogVisible={() => {}}
        saveCropImage={() => {}}
        callCroppedImage={mockCallCroppedImage}
        shadingColor="black"
        backgroundColor="white"
        buttonName="Save"
        modalTitle="Crop Image"
      />
    );

    const avatarComponent = getByTestId("avatarMock");
    fireEvent.click(avatarComponent);
    waitFor(() => {
      expect(mockCallCroppedImage).toHaveBeenCalled();
    });
  });
});

describe("Snapshot for SentaraImageCropper", () => {
  it("takes snapShot of the component", () => {
    const mockSaveCropImage = jest.fn();
    expect(
      <SentaraImageCropper
        imageCrop={true}
        isDialogVisible={() => {}}
        saveCropImage={mockSaveCropImage}
        callCroppedImage={() => {}}
        shadingColor="black"
        backgroundColor="white"
        buttonName="Save"
        modalTitle="Crop Image"
      />
    ).toMatchSnapshot();
  });
});
