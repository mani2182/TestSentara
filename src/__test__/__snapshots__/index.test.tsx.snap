// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for SentaraImageCropper takes snapShot of the component 1`] = `
<SentaraImageCropper
  backgroundColor="white"
  buttonName="Save"
  callCroppedImage={[Function]}
  imageCrop={true}
  isDialogVisible={[Function]}
  modalTitle="Crop Image"
  saveCropImage={[MockFunction]}
  shadingColor="black"
/>
`;
